### ISSUE TITLE
Classpath example fails against Accumulo 3.x

### ISSUE THREAD

**Issue by jmark99** [apache/accumulo-examples#111]
The Classpath example fails when attempting to set the FooFilter iterator.

`setiter -n foofilter -p 10 -scan -minc -majc -class org.apache.accumulo.test.FooFilter`

**Comment by Manno15** [comment]
@jmark99 Which version of hadoop were you using? I don't see anything obvious that has changed from this example and from the accumulo code it uses to get the class loader but I will keep looking.

**Comment by Manno15** [comment]
Some more info from server logs: 

<details>

```
Failed to load class org.apache.accumulo.core.spi.compaction.SimpleCompactionDispatcher in classloader context cx1
java.io.UncheckedIOException: Error getting context class loader for context: cx1
	at org.apache.accumulo.start.classloader.vfs.AccumuloVFSClassLoader.getContextClassLoader(AccumuloVFSClassLoader.java:422)
	at org.apache.accumulo.core.classloader.DefaultContextClassLoaderFactory.getClassLoader(DefaultContextClassLoaderFactory.java:99)
	at org.apache.accumulo.core.classloader.ClassLoaderUtil.getClassLoader(ClassLoaderUtil.java:77)
	at org.apache.accumulo.core.classloader.ClassLoaderUtil.loadClass(ClassLoaderUtil.java:85)
	at org.apache.accumulo.core.conf.ConfigurationTypeHelper.getClassInstance(ConfigurationTypeHelper.java:209)
	at org.apache.accumulo.core.conf.ConfigurationTypeHelper.getClassInstance(ConfigurationTypeHelper.java:182)
	at org.apache.accumulo.core.conf.Property.createTableInstanceFromPropertyName(Property.java:1841)
	at org.apache.accumulo.server.conf.TableConfiguration.createCompactionDispatcher(TableConfiguration.java:192)
	at org.apache.accumulo.server.conf.TableConfiguration.lambda$new$2(TableConfiguration.java:78)
	at org.apache.accumulo.core.conf.AccumuloConfiguration$DeriverImpl.derive(AccumuloConfiguration.java:529)
	at org.apache.accumulo.server.conf.TableConfiguration.getCompactionDispatcher(TableConfiguration.java:226)
	at org.apache.accumulo.tserver.tablet.CompactableImpl.getConfiguredService(CompactableImpl.java:1456)
	at org.apache.accumulo.tserver.tablet.CompactableImpl.lambda$new$7(CompactableImpl.java:698)
	at com.google.common.base.Suppliers$ExpiringMemoizingSupplier.get(Suppliers.java:262)
	at org.apache.accumulo.tserver.tablet.CompactableImpl.isMajorCompactionQueued(CompactableImpl.java:1517)
	at org.apache.accumulo.tserver.tablet.Tablet.isMajorCompactionQueued(Tablet.java:1396)
	at org.apache.accumulo.tserver.TabletServer.lambda$getStats$7(TabletServer.java:1092)
	at com.google.common.collect.ImmutableSortedMap.forEach(ImmutableSortedMap.java:793)
	at org.apache.accumulo.tserver.TabletServer.getStats(TabletServer.java:1061)
	at org.apache.accumulo.tserver.TabletClientHandler.getTabletServerStatus(TabletClientHandler.java:1016)
	at jdk.internal.reflect.GeneratedMethodAccessor3.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.apache.accumulo.core.trace.TraceUtil.lambda$wrapService$0(TraceUtil.java:206)
	at com.sun.proxy.$Proxy40.getTabletServerStatus(Unknown Source)
	at org.apache.accumulo.core.tabletserver.thrift.TabletClientService$Processor$getTabletServerStatus.getResult(TabletClientService.java:3016)
	at org.apache.accumulo.core.tabletserver.thrift.TabletClientService$Processor$getTabletServerStatus.getResult(TabletClientService.java:2992)
	at org.apache.thrift.ProcessFunction.process(ProcessFunction.java:40)
	at org.apache.thrift.TBaseProcessor.process(TBaseProcessor.java:40)
	at org.apache.thrift.TMultiplexedProcessor.process(TMultiplexedProcessor.java:147)
	at org.apache.accumulo.server.rpc.TimedProcessor.process(TimedProcessor.java:54)
	at org.apache.thrift.server.AbstractNonblockingServer$FrameBuffer.invoke(AbstractNonblockingServer.java:492)
	at org.apache.accumulo.server.rpc.CustomNonBlockingServer$CustomFrameBuffer.invoke(CustomNonBlockingServer.java:129)
	at org.apache.thrift.server.Invocation.run(Invocation.java:18)
	at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.apache.commons.vfs2.FileSystemException: Could not determine the type of file "hdfs://localhost:9870/user1/lib/[^.].*.jar".
	at org.apache.commons.vfs2.provider.AbstractFileObject.attach(AbstractFileObject.java:171)
	at org.apache.commons.vfs2.provider.AbstractFileObject.getType(AbstractFileObject.java:1370)
	at org.apache.accumulo.start.classloader.vfs.AccumuloVFSClassLoader.resolve(AccumuloVFSClassLoader.java:145)
	at org.apache.accumulo.start.classloader.vfs.AccumuloReloadingVFSClassLoader.<init>(AccumuloReloadingVFSClassLoader.java:187)
	at org.apache.accumulo.start.classloader.vfs.AccumuloReloadingVFSClassLoader.<init>(AccumuloReloadingVFSClassLoader.java:259)
	at org.apache.accumulo.start.classloader.vfs.ContextManager$Context.getClassLoader(ContextManager.java:56)
	at org.apache.accumulo.start.classloader.vfs.ContextManager.getClassLoader(ContextManager.java:188)
	at org.apache.accumulo.start.classloader.vfs.AccumuloVFSClassLoader.getContextClassLoader(AccumuloVFSClassLoader.java:420)
	... 38 more
Caused by: java.io.IOException: Failed on local exception: org.apache.hadoop.ipc.RpcException: RPC response exceeds maximum data length; Host Details : local host is: "groot/192.168.50.248"; destination host is: "localhost":9870; 
	at org.apache.hadoop.net.NetUtils.wrapException(NetUtils.java:808)
	at org.apache.hadoop.ipc.Client.getRpcResponse(Client.java:1495)
	at org.apache.hadoop.ipc.Client.call(Client.java:1437)
	at org.apache.hadoop.ipc.Client.call(Client.java:1347)
	at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:228)
	at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:116)
	at com.sun.proxy.$Proxy36.getFileInfo(Unknown Source)
	at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.getFileInfo(ClientNamenodeProtocolTranslatorPB.java:876)
	at jdk.internal.reflect.GeneratedMethodAccessor6.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:422)
	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeMethod(RetryInvocationHandler.java:165)
	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invoke(RetryInvocationHandler.java:157)
	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeOnce(RetryInvocationHandler.java:95)
	at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:359)
	at com.sun.proxy.$Proxy37.getFileInfo(Unknown Source)
	at org.apache.hadoop.hdfs.DFSClient.getFileInfo(DFSClient.java:1619)
	at org.apache.hadoop.hdfs.DistributedFileSystem$29.doCall(DistributedFileSystem.java:1495)
	at org.apache.hadoop.hdfs.DistributedFileSystem$29.doCall(DistributedFileSystem.java:1492)
	at org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81)
	at org.apache.hadoop.hdfs.DistributedFileSystem.getFileStatus(DistributedFileSystem.java:1507)
	at org.apache.commons.vfs2.provider.hdfs.HdfsFileObject.doAttach(HdfsFileObject.java:90)
	at org.apache.commons.vfs2.provider.AbstractFileObject.attach(AbstractFileObject.java:162)
	... 45 more
Caused by: org.apache.hadoop.ipc.RpcException: RPC response exceeds maximum data length
	at org.apache.hadoop.ipc.Client$IpcStreams.readResponse(Client.java:1810)
	at org.apache.hadoop.ipc.Client$Connection.receiveRpcResponse(Client.java:1165)
	at org.apache.hadoop.ipc.Client$Connection.run(Client.java:1061)

```


</details>

Also see multiple dispatch compaction errors: 

<details>

```
Failed to dispatch compaction 2<< kind:SYSTEM hints:null, falling back to default service.
java.lang.NullPointerException
	at org.apache.accumulo.server.conf.TableConfiguration.createCompactionDispatcher(TableConfiguration.java:198)
	at org.apache.accumulo.server.conf.TableConfiguration.lambda$new$2(TableConfiguration.java:78)
	at org.apache.accumulo.core.conf.AccumuloConfiguration$DeriverImpl.derive(AccumuloConfiguration.java:529)
	at org.apache.accumulo.server.conf.TableConfiguration.getCompactionDispatcher(TableConfiguration.java:226)
	at org.apache.accumulo.tserver.tablet.CompactableImpl.getConfiguredService(CompactableImpl.java:1456)
	at org.apache.accumulo.tserver.tablet.CompactableImpl.lambda$new$7(CompactableImpl.java:698)
	at com.google.common.base.Suppliers$ExpiringMemoizingSupplier.get(Suppliers.java:262)
	at org.apache.accumulo.tserver.tablet.CompactableImpl.isMajorCompactionQueued(CompactableImpl.java:1517)
	at org.apache.accumulo.tserver.tablet.Tablet.isMajorCompactionQueued(Tablet.java:1396)
	at org.apache.accumulo.tserver.TabletServer.lambda$getStats$7(TabletServer.java:1092)
	at com.google.common.collect.ImmutableSortedMap.forEach(ImmutableSortedMap.java:793)
	at org.apache.accumulo.tserver.TabletServer.getStats(TabletServer.java:1061)
	at org.apache.accumulo.tserver.TabletClientHandler.getTabletServerStatus(TabletClientHandler.java:1016)
	at jdk.internal.reflect.GeneratedMethodAccessor3.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.apache.accumulo.core.trace.TraceUtil.lambda$wrapService$0(TraceUtil.java:206)
	at com.sun.proxy.$Proxy38.getTabletServerStatus(Unknown Source)
	at org.apache.accumulo.core.tabletserver.thrift.TabletClientService$Processor$getTabletServerStatus.getResult(TabletClientService.java:3016)
	at org.apache.accumulo.core.tabletserver.thrift.TabletClientService$Processor$getTabletServerStatus.getResult(TabletClientService.java:2992)
	at org.apache.thrift.ProcessFunction.process(ProcessFunction.java:40)
	at org.apache.thrift.TBaseProcessor.process(TBaseProcessor.java:40)
	at org.apache.thrift.TMultiplexedProcessor.process(TMultiplexedProcessor.java:147)
	at org.apache.accumulo.server.rpc.TimedProcessor.process(TimedProcessor.java:54)
	at org.apache.thrift.server.AbstractNonblockingServer$FrameBuffer.invoke(AbstractNonblockingServer.java:492)
	at org.apache.accumulo.server.rpc.CustomNonBlockingServer$CustomFrameBuffer.invoke(CustomNonBlockingServer.java:129)
	at org.apache.thrift.server.Invocation.run(Invocation.java:18)
	at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52)
	at java.base/java.lang.Thread.run(Thread.java:829)


```
</details?

**Comment by Manno15** [comment]
I am also seeing the same issue against accumulo 2.0.1 with hadoop 3.3.4 (also saw this with other versions of hadoop) and zookeeper 3.8.0

**Comment by jmark99** [comment]
@Manno15 I was using Hadoop 3.3.4 and ZK 3.8.0. As you, I'm seeing the same issue with the older 2.x versions. I know that it has worked in the past, so not sure what has changed to cause the error yet.

**Comment by ctubbsii** [comment]
The error message `Caused by: org.apache.commons.vfs2.FileSystemException: Could not determine the type of file "hdfs://localhost:9870/user1/lib/[^.].*.jar".` seems similar to what we saw in apache/accumulo#2682 and apache/accumulo#3069. However, in this case, it could be some issue with the file globbing to expand the pattern. Does the directory exist? Is it readable by the user running the Accumulo process?

**Comment by Manno15** [comment]
The errors seem to be from the command `config -t examples.nofoo -s table.class.loader.context=cx1` which just proceeds the `setiter` command. The example may be missing some steps on setting the context. Then this will cause the `setiter` command to fail since it cannot get the context class loader. 

I am still unsure on the `Caused by: org.apache.commons.vfs2.FileSystemException: Could not determine the type of file "hdfs://localhost:9870/user1/lib/[^.].*.jar".` This will be thrown even when specifying the full path to the jar file and I cannot tell if its because of the issues you mentioned or just because the context classloader part failed.

**Comment by Manno15** [comment]
Scratch my previous comment. I got the example to work on my latest run. The issue resided in using the wrong host and port in this command: `config -s general.vfs.context.classpath.cx1=hdfs://<namenode host>:<namenode port>/user1/lib/[^.].*.jar`. I also specified FooFilter instead of relying on it to interpret the pattern fully. The rest of the example worked as expected. 

I used the host:port set with this property: fs.default.name
